chips tests fail when sherpa-test-data is not initialized

# Summary

The chips tests probably need a `@requires_data` decorator added (see PR #266). This is not a show stopper for the 4.8.2 release.

# Details

The ChIPS tests fail if sherpa-test-data is not initialized (tested with both python 2.7 and 3.5). This is on the master branch and astropy and matplotlib are installed, python-2.7.

```
% git rev-parse HEAD
770359b5004374b969ebb63c173f293419397b4c
% git submodule init
....
% git submodule update
Submodule path 'sherpa-test-data': checked out '488dd7e1873067ba461726f4032b473b653e11f7'
% python setup.py test -a sherpa/astro/ui/tests/test_chips_unit.py
...
... tests pass ...
...
============================ pytest-warning summary ============================
WC1 None passing a string to pytest.main() is deprecated, pass a list of arguments instead.
=========== 17 passed, 1 xfailed, 1 pytest-warnings in 9.39 seconds ============
```

Now if I remove the sherpa-test-data directory we get failures

```
% git submodule deinit sherpa-test-data
% python setup.py test -a sherpa/astro/ui/tests/test_chips_unit.py
...
collected 18 items 

sherpa/astro/ui/tests/test_chips_unit.py::test_plot[model-args0] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[arf-args1] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[source-args2] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg-args3] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_model-args4] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_resid-args5] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_ratio-args6] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_delchi-args7] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_chisqr-args8] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_fit-args9] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_source-args10] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[energy_flux-args11] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[photon_flux-args12] xfail
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_fit_resid-args13] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[bkg_fit_delchi-args14] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[source_component-args15] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[model_component-args16] ERROR
sherpa/astro/ui/tests/test_chips_unit.py::test_plot[order-args17] ERROR

==================================== ERRORS ====================================
___________________ ERROR at setup of test_plot[model-args0] ___________________

a = None, p = ('3c273.pi',), sep = '/', path = None, b = '3c273.pi'

    def join(a, *p):
        """Join two or more pathname components, inserting '/' as needed.
        If any component is an absolute path, all previous path components
        will be discarded.  An empty last part will result in a path that
        ends with a separator."""
        sep = _get_sep(a)
        path = a
        try:
            if not p:
                path[:0] + sep  #23780: Ensure compatible data type even if p is null.
            for b in p:
                if b.startswith(sep):
                    path = b
                elif not path or path.endswith(sep):
>                   path += b
E                   TypeError: unsupported operand type(s) for +=: 'NoneType' and 'str'

../../miniconda2/envs/sherpa-stat-changes/lib/python3.5/posixpath.py:85: TypeError

During handling of the above exception, another exception occurred:

ui = <module 'sherpa.astro.ui' from '/home/djburke/sherpa/sherpa-stat-changes/sherpa/astro/ui/__init__.py'>
make_data_path = <function make_data_path.<locals>.wrapped at 0x7f1565288048>

    @requires_fits
    @pytest.fixture(autouse=True)
    def load_data(ui, make_data_path):
        """
        Load dataset before every test.
        """
>       ui.load_data(make_data_path("3c273.pi"))

```

This seems to be the error on all the cases.

--- a/sherpa/astro/ui/tests/test_chips_unit.py
+++ b/sherpa/astro/ui/tests/test_chips_unit.py
@@ -17,7 +17,7 @@
 #  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 import pytest
-from sherpa.utils import requires_fits
+from sherpa.utils import requires_data, requires_fits
 from six.moves import reload_module
 
 
@@ -57,6 +57,7 @@ def clean_up(request, ui):
     request.addfinalizer(fin)
 
 
+@requires_data
 @requires_fits
 @pytest.fixture(autouse=True)
 def load_data(ui, make_data_path):
@@ -102,6 +103,3 @@ def test_plot(call, args, chips, ui):
     function = getattr(ui, "plot_"+call)
     function(*args)
     assert_chips_called(chips)
-
-
-

--- a/sherpa/conftest.py
+++ b/sherpa/conftest.py
@@ -132,11 +132,17 @@ def make_data_path():
 
     Returns
     -------
-    make_data_path : func A function that accepts a list of path elements to be joined with the base data dir path
+    make_data_path : func
+        A function that accepts a list of path elements to be joined with
+        the base data dir path. This function exits with a RuntimeError
+        if the data directory is None, pointing out the requires_data
+        decorator is needed.
     """
     path = SherpaTestCase.datadir
 
     def wrapped(arg):
+        if path is None:
+            raise RuntimeError("Test needs the requires_data decorator")
         return os.path.join(path, arg)
 
     return wrapped

